utils/codepointtrie/benches/codepointtrie.rs

```diff
@@ -0,0 +1,80 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+use criterion::{black_box, criterion_group, criterion_main, Criterion};
+
+use icu_codepointtrie::toml::CodePointTrieToml;
+use icu_codepointtrie::CodePointTrie;
+use std::convert::TryInto;
+use std::fs;
+
+#[cfg(feature = "bench")]
+mod sample_str_lng {
+    pub const ENG: &str = "Universal Declaration of Human Rights";
+    pub const PCD: &str = "Dèclaråcion dès dreûts d' l'ome po tos lès payîs dè monde";
+    pub const UKR: &str = "ЗАГАЛЬНА ДЕКЛАРАЦІЯ ПРАВ ЛЮДИНІ";
+    pub const YUE: &str = "世界人权宣言";
+    pub const CCP: &str = "𑄟𑄚𑄬𑄭𑄃𑄇𑄴𑄇𑄥𑄧𑄁𑄢𑄴 𑄝𑄬𑄇𑄴𑄅𑄚𑄮𑄢𑄴 𑄟𑄧𑄚𑄳𑄢𑄧𑄧𑄇𑄉𑄮𑄌𑄴";
+}
+
+const SAMPLE_STRING_MIXED: &str = "Dèclaråcion ЗАГАЛЬНА 世界人权宣言 𑄟𑄚𑄬𑄭𑄃𑄇𑄴𑄇𑄥𑄧𑄁𑄢𑄴";
+
+/// A function that returns 100 code points in the desired language
+fn one_hundred_code_points(sample_str: &str) -> String {
+    sample_str.chars().cycle().take(100).collect()
+}
+
+fn load_code_point_trie(buffer: &mut Vec<u8>) -> CodePointTrie<u8> {
+    let path = concat!(
+        env!("CARGO_MANIFEST_DIR"),
+        "/benches/testdata/gc_for_test.toml"
+    );
+    *buffer = fs::read(path).unwrap();
+    let toml: CodePointTrieToml = toml::from_slice(buffer.as_slice()).unwrap();
+    (&toml).try_into().unwrap()
+}
+
+fn overview_bench(c: &mut Criterion) {
+    let s = one_hundred_code_points(SAMPLE_STRING_MIXED);
+    let mut buffer = Vec::<u8>::new();
+    let cpt = load_code_point_trie(&mut buffer);
+
+    c.bench_function("cpt/overview", |b| {
+        b.iter(|| {
+            black_box(&s)
+                .chars()
+                .map(|c| black_box(&cpt).get(c as u32))
+                .reduce(|s, v| s.wrapping_add(v))
```

Comment:
suggestion: rename closure local `s` to avoid confusion

It looks like the `s` on lines 39 and 45 is different than the `s` here on line 48, but it's slightly confusing.  If the `s` here is a throwaway sum to make use of the result of each code point's `u8` return value, then maybe `sum`?  My preference is for `some_sum` :-)

Reply:
Yeah, `s` is just the variable name I usually use as the accumulator in `reduce`, but it's a bad choice here because it shadows another variable.